pkg/operator/ceph/object/controller.go

```diff
@@ -474,10 +474,10 @@ func (r *ReconcileCephObjectStore) startMonitoring(objectstore *cephv1.CephObjec
 
 	var port string
 
-	if objectstore.Spec.Gateway.Port != 0 {
-		port = strconv.Itoa(int(objectstore.Spec.Gateway.Port))
-	} else if objectstore.Spec.Gateway.SecurePort != 0 {
+	if objectstore.Spec.Gateway.SecurePort != 0 && objectstore.Spec.Gateway.SSLCertificateRef != "" {
 		port = strconv.Itoa(int(objectstore.Spec.Gateway.SecurePort))
+	} else if objectstore.Spec.Gateway.Port != 0 {
+		port = strconv.Itoa(int(objectstore.Spec.Gateway.Port))
 	}
```

Comment:
If both port and securePort are 0, this is a good error condition to check in the admission controller (for a separate PR), and also we should log the error here.